ibm/service/iamaccessgroup/resource_ibm_iam_access_group_template_assignment.go

```diff
@@ -0,0 +1,402 @@
+	}
+	if err = d.Set("target", templateAssignmentVerboseResponse.Target); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting target: %s", err))
+	}
+	if err = d.Set("account_id", templateAssignmentVerboseResponse.AccountID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting account_id: %s", err))
+	}
+	if err = d.Set("operation", templateAssignmentVerboseResponse.Operation); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting operation: %s", err))
+	}
+	if err = d.Set("status", templateAssignmentVerboseResponse.Status); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting status: %s", err))
+	}
+	if err = d.Set("href", templateAssignmentVerboseResponse.Href); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting href: %s", err))
+	}
+	if err = d.Set("created_at", flex.DateTimeToString(templateAssignmentVerboseResponse.CreatedAt)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting created_at: %s", err))
+	}
+	if err = d.Set("created_by_id", templateAssignmentVerboseResponse.CreatedByID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting created_by_id: %s", err))
+	}
+	if err = d.Set("last_modified_at", flex.DateTimeToString(templateAssignmentVerboseResponse.LastModifiedAt)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting last_modified_at: %s", err))
+	}
+	if err = d.Set("last_modified_by_id", templateAssignmentVerboseResponse.LastModifiedByID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting last_modified_by_id: %s", err))
+	}
+	if err = d.Set("etag", response.Headers.Get("Etag")); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting etag: %s", err))
+	}
+
+	return nil
+}
+
+func resourceIBMIamAccessGroupTemplateAssignmentUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	fmt.Println("==========update===================")
+	iamAccessGroupsClient, err := meta.(conns.ClientSession).IAMAccessGroupsV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateAssignmentOptions := &iamaccessgroupsv2.UpdateAssignmentOptions{}
+
+	updateAssignmentOptions.SetAssignmentID(d.Id())
+
+	getAssignmentOptions := &iamaccessgroupsv2.GetAssignmentOptions{}
+
+	getAssignmentOptions.SetAssignmentID(d.Id())
+
+	templateAssignmentVerboseResponse, response, err := iamAccessGroupsClient.GetAssignmentWithContext(context, getAssignmentOptions)
+
+	if err != nil && templateAssignmentVerboseResponse == nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetAssignmentWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetAssignmentWithContext failed %s\n%s", err, response))
+	}
+
+	updateAssignmentOptions.SetIfMatch(response.Headers.Get("ETag"))
+
+	hasChange := false
+
+	if d.HasChange("template_version") {
+		fmt.Println("==========update1===================")
```

Comment:
Only "template_version" can be updated . Remaining arguments in schema should we add ForceNew on them ?

Reply:
As discussed, retaining the same behaviour.